### DISCUSSION TITLE
ESP32-S3-EYE support

### DISCUSSION THREAD

**Discussion by alflexRH** [jomjol/AI-on-the-edge-device#779] [Ideas]
At the moment there seems to be a limit of taking 1 picture every ~2 minutes. Would using an ESP32-S3 improve that number much? Espressif does boast about the AI acceleration :)

Would be more then happy to arrange 1 or 2 ESP32-S3-EYE modules for developers if that helps.

**Comment by jomjol** [comment]
If I have a look to the comparison (https://gist.github.com/sekcompsci/2bf39e715d5fe47579fa184fa819f421), it looks like there is an upgraded processer (LX7), which might have some advantages.

If your request is a faster cycle time, this can be achieved also on the ESP32-CAM with some tricks (improved alignment, fast dedicated neural networks). With this you can go down well below 1 minute.

If the ESP-S3 is a further improvement, would need a test, but currently I don't have one.

**Comment by alflexRH** [comment]
I think the biggest gains can be obtained with this: https://github.com/espressif/esp-nn , which makes use of the new vector instructions found in the ESP32-S3. 

but the S3 is still very new. i dont think there are any chinese boards yet. just the Espressif original dev boards, but they are in stock at Mouser https://eu.mouser.com/ProductDetail/Espressif-Systems/ESP32-S3-EYE

**Reply by caco3** [reply]
They are quite more expensive!
And they have a display which is not really needed.

For normal monitoring purpose, I think 3 minutes interval is fast enough, at least for my usage (gas, water)

**Comment by jomjol** [comment]
You are right. But at Mouser it is about EUR 44,-. I will wait until it will get much cheaper. No reason to change currently.

BTW: just checked quickly the schematics. It looks like EVERY GPIO is occupied. Therefore: how could you connect the external LED? One free GPIO needed!

**Comment by alflexRH** [comment]
Plenty of options to get a free IO.
*) if you dont use the 1.3" LCD screen, that frees up 5 of them. 
*) de-solder the battery voltage divider will give you 1 IO
*) the lines going to the microphone are nicely routed via series resistors that can be removed, giving 3 IOs
*) i think you can use the strapping pins GPIO45 and GPIO46 as outputs after boot

or an I2C led driver / IO expander if you wana go all fancy. 

Very tempted to design a custom board for it ;) could even skip the SD card if you get an ESP32-S3 module with 8 or 16MB onboard flash. all round, camera centered, led ring light hehe

**Reply by jomjol** [reply]
You are totally right, but the USP of the AI-on-the-Edge is:
1) cheap hardware (10 EUR)
2) very easy setup: just hang it above the counter - no soldering, ...
Therefore, almost everybody with rudimentary computer skills is able to us this - resulting in the high interest.#

Imaging to go to the ESP32-S3-EYE:
1. 4x price increase
2. you need to desolder and solder compoments
That would end up in never ending complains about not working / broken system,  ... .

For commercial applications this looks totally different.

**Comment by alflexRH** [comment]
Lilygo now has a ESP32-S3 camera module. at €20 not quite at the €10 level, but getting there! no white LED tho :(

https://www.aliexpress.com/item/1005004551008036.html

**Comment by Adminius** [comment]
under 13$ with Camera, S3 and SD-Card and USB Port
https://www.aliexpress.com/item/1005006661339895.html

**Comment by argetlam-coder** [comment]
I have a Freenove ESP32-S3-WROOM board. On my board, the processing time for an image is approx. 10s.

I have a [branch](https://github.com/argetlam-coder/AI-on-the-edge-device/tree/add-freenove-board-support) which contains the changes necessary to get the software to run on this board. However, the camera no longer works for me since I applied the last changes from the upstream project.

I have another [branch](https://github.com/argetlam-coder/AI-on-the-edge-device/tree/add-esp32-s3-support) which only contains the changes that are necessary for the software to compile for the ESP32-S3.

**Comment by schrej-zz** [comment]
The ESP32-S3-WROOM incl. cam is now below 10€.
Would you be able to compile it now?

**Comment by martin8310** [comment]
any news here?

**Reply by Adminius** [reply]
https://github.com/Slider0007/AI-on-the-edge-device

**Comment by Slider0007** [comment]
Fork with support for various ESP32 and ESP32-S3 boards --> more infos: #3874
